### ISSUE TITLE
Add documentation for the PhoneInput props with description

### ISSUE THREAD

**Issue by rodrigomf24** [catamphetamine/react-phone-number-input#191]
Hi,

After breaking my head on why the `PhoneInput` didn't supported limiting characters(having such a robust feature of matching the phone with the selected country and all...), I found out by reading the source code that there was a prop to enable such feature. 

It would be good to add to the README the description of the allowed props like the way you have them on the type definition of the component class, this can save others some research time...

https://github.com/catamphetamine/react-phone-number-input/blob/master/source/PhoneInput.js#L200

By the way thanks for making such an awesome component 🍺

**Comment by catamphetamine** [comment]
Well, actually, there was no length limiting feature at all.
I just added that property recently because several people asked about that.
I myself don't see a reason for limiting input max length, and if I added it to the readme I guess I'd be advocating its use which I don't.
On the other hand, as you said, it's not easy to find out about this property.
So either of the possible solution isn't ideal.
